pynestml/symbols/template_type_symbol.py

```diff
@@ -0,0 +1,49 @@
+#
+# template_type_symbol.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+from pynestml.symbols.type_symbol import TypeSymbol
+
+
+class TemplateTypeSymbol(TypeSymbol):
+    def __init__(self, i):
+        super(TemplateTypeSymbol, self).__init__(name='_template_' + str(i))
+        self.i = i
```

Comment:
I would make this name more explicit (and potentially "private", i.e. ``_integer_type`` with an ``@property`` function)